rerun_cpp/src/rerun/time_column.hpp

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include <cassert>
+#include <chrono>
+#include <memory> // shared_ptr
+
+#include "collection.hpp"
+#include "error.hpp"
+#include "timeline.hpp"
+
+struct rr_time_column;
+
+namespace arrow {
+    class Array;
+}
+
+namespace rerun {
+    /// Describes whether an array is known to be sorted or not.
+    enum class SortingStatus {
+        /// It's not known whether the array is sorted or not.
+        Unknown = 0,
+        /// The array is known to be sorted.
+        Sorted = 1,
+        /// The array is known to be unsorted.
+        Unsorted = 2,
+    };
+
+    /// Arrow-encoded data for a column of time points.
+    ///
+    /// \see `rerun::RecordingStream::send_columns`
+    struct TimeColumn {
+        /// The timeline this column belongs to.
+        Timeline timeline;
+
+        /// Time points as a primitive array of i64.
+        std::shared_ptr<arrow::Array> array;
+
+        /// The sorting order of the `times` array.
+        SortingStatus sorting_status;
+
+      public:
+        /// Creates a new time column from an array of time points.
+        ///
+        /// \param timeline The timeline this column belongs to.
+        /// \param timepoints The time points.
+        /// Depending on the `TimeType` of the timeline this may be either timestamps or sequence numbers.
+        /// Make sure the sorting status is correctly specified.
+        /// \param sorting_status The sorting status of the time points.
+        TimeColumn(
+            Timeline timeline, Collection<int64_t> timepoints,
+            SortingStatus sorting_status = SortingStatus::Sorted
```

Comment:
```suggestion
            SortingStatus sorting_status = SortingStatus::Unknown
```

This seems like a much safer choice, here and everywhere